core/variant_call.cpp

```diff
@@ -215,6 +215,8 @@ struct _VariantCall {
 	static void _call_##m_type##_##m_method(Variant &r_ret, Variant &p_self, const Variant **p_args) { reinterpret_cast<m_type *>(p_self._data._mem)->m_method(); }
 #define VCALL_LOCALMEM0R(m_type, m_method) \
 	static void _call_##m_type##_##m_method(Variant &r_ret, Variant &p_self, const Variant **p_args) { r_ret = reinterpret_cast<m_type *>(p_self._data._mem)->m_method(); }
+#define VCALL_LOCALMEM0RI(m_type, m_method, m_internal_method) \
+	static void _call_##m_type##_##m_method(Variant &r_ret, Variant &p_self, const Variant **p_args) { r_ret = reinterpret_cast<m_type *>(p_self._data._mem)->m_internal_method(); }
```

Comment:
That's a bit weird IMO and unexpected that `to_int` in the scripting API will not call `to_int` in the core API. Needs discussion with @godotengine/core folks to see what's best.

Another option would be to just change `to_int()` to be 64-bit like in `master`. It would break compatibility but this change also breaks compatibility for the scripting API while introducing inconsistency.